### PR TITLE
RC_Channel: remove include of RC_Channel_aux at end of RC_Channel.h

### PR DESCRIPTION
**Pull request by kwikius**
RC_Channel: remove include of RC_Channel_aux at end of RC_Channel.h and include RC_Channel_aux.h where required.

### CODE REVIEW

File: libraries/RC_Channel/RC_Channel.h [hunk 1]
```diff
@@ -40,7 +40,7 @@ class RC_Channel {
     };
 
     // setup min and max radio values in CLI
-    void        update_min_max();
+    void        update_min_max();  //
```
Comment (OXINARF) [inline libraries/RC_Channel/RC_Channel.h]: Leftover?

File: libraries/RC_Channel/RC_Channel.cpp [hunk 2]
```diff
@@ -306,7 +306,7 @@ RC_Channel::update_min_max()
   the current radio_in value using the specified dead_zone
  */
 int16_t
-RC_Channel::pwm_to_angle_dz_trim(uint16_t dead_zone, uint16_t _trim)
+RC_Channel::pwm_to_angle_dz_trim(uint16_t dead_zone, uint16_t _trim)const
```
Comment (OXINARF) [inline libraries/RC_Channel/RC_Channel.cpp]: I'm not sure if there's a specific rule for this, but I think that a space before const would be appreciated

File: libraries/RC_Channel/RC_Channel.h [hunk 3]
```diff
@@ -83,9 +83,9 @@ class RC_Channel {
     // generate PWM from servo_out value
     void        calc_pwm(void);
 
-    int16_t     pwm_to_angle_dz_trim(uint16_t dead_zone, uint16_t trim);
-    int16_t     pwm_to_angle_dz(uint16_t dead_zone);
-    int16_t     pwm_to_angle();
+    int16_t     pwm_to_angle_dz_trim(uint16_t dead_zone, uint16_t trim) const;
```
Comment (OXINARF) [inline libraries/RC_Channel/RC_Channel.h]: Ok, here you have a space. So please make them all consistent - with the space.

File: libraries/RC_Channel/RC_Channel.h [hunk 4]
```diff
@@ -157,9 +157,24 @@ class RC_Channel {
...
-    int16_t     _radio_in;
-    // value generated from PWM
+    // The raw PWM input from user joystick. Range approx 900 millisec to 2100 millisec [see <AP_HAL/RCInput.h>]
+    // set by 
+    //    this->set_radio_in(int16_t val)  // only used by arduplane at startup
+    //    this->set_pwm(int16_t pwm) 
+    //    this->set_pwm_no_deadzone(int16_t pwm)
+    //    this->input()  // which  calls the Hal::RCInput abstraction, which is the runtime update
+    int16_t     _radio_in;  
+   //-------------------------------------
```
Comment (OXINARF) [inline libraries/RC_Channel/RC_Channel.h]: I think this line isn't needed

**Comment by kwikius** [comment]
Good points. The comments are more for my benefit than anything. I will close it and reopen it when I have more work on it

**Comment by OXINARF** [comment]
@kwikius The comments are very valuable.

**Pull request closed without merging**